src/coreservices.cpp

```diff
@@ -123,7 +124,14 @@ CoreServices::CoreServices(const CmdlineArgs& args, QApplication* pApp)
     initializeKeyboard();
 }
 
-CoreServices::~CoreServices() = default;
+CoreServices::~CoreServices() {
+    if (!m_isInitialized) {
+        qDebug() << "Skipping CoreServices shutdown because it was never initialized.";
+        return;
+    }
+
+    shutdown();
```

Comment:
How about removing the shutdown function and moving its implementation to the destructor?

Reply:
Well, maybe we could keep both methods separate, because the belong to each other. One does the *set up* and its sibling the *tear down*. How about renaming them to `initialize()` and `finalize()` to emphasize this relationship.

Reply:
I removed the method. Originally I wanted to keep it to keep the line count small.